### PR TITLE
change system_program to use instruction indexes instead of hard-coded ones

### LINKED ISSUE #1551: Can't batch System instructions
SystemProgram ignores the instruction's `accounts` field and accesses the Transaction's `account_keys` field directly. That means, for example, you can't batch two independent `Move` instructions in one transaction.

### PR DESCRIPTION
**Pull request by rob-solana**
Fixes #1551

### CODE REVIEW

**Comment by garious** [comment]
Nice! That change actually lights up a more impressive test. Add a "crowdfund" transaction type and a test that demos multiple **signed** `from` keys.

File: src/system_program.rs [hunk 1]
```diff
@@ -248,6 +265,22 @@ mod test {
         assert_eq!(accounts[0].tokens, 0);
         assert_eq!(accounts[1].tokens, 1);
     }
+
+    #[test]
+    fn test_move_many() {
+        use logger;
+        logger::setup();
+        let from = Keypair::new();
+        let tos = [(Keypair::new().pubkey(), 1), (Keypair::new().pubkey(), 1)];
```
Comment (sakridge) [inline src/system_program.rs]: unique amounts might catch ordering bugs

File: src/system_program.rs [hunk 2]
```diff
@@ -123,7 +136,11 @@ mod test {
 
     fn process_transaction(tx: &Transaction, accounts: &mut [Account]) -> Result<()> {
         let mut refs: Vec<&mut Account> = accounts.iter_mut().collect();
-        super::process_instruction(&tx, 0, &mut refs[..])
+
+        for (idx, _) in tx.instructions.iter().enumerate() {
```
Comment (garious) [inline src/system_program.rs]: `for idx in 0..tx.instructions.len()`

**Comment by garious** [comment]
Yeah, you're right. Only remaining problem I see is no check that key 0 is signed.

**Pull request closed without merging**